### PR TITLE
br: make logs more clear.

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Problem Summary:

### What is changed and how it works?
make log more clear
1. add warn log when set tiflash replica to nil during restore
2. make retryable error to warn log level.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -1972,6 +1972,13 @@ func (rc *Client) PreCheckTableTiFlashReplica(
 			// we cannot satisfy TiFlash replica in restore cluster. so we should
 			// set TiFlashReplica to unavailable in tableInfo, to avoid TiDB cannot sense TiFlash and make plan to TiFlash
 			// see details at https://github.com/pingcap/br/issues/931
+			// TODO maybe set table.Info.TiFlashReplica.Count to tiFlashStoreCount, but we need more test about it.
+			log.Warn("Table does not satisfy tiflash replica requirements, set tiflash replcia to unavaiable",
+				zap.Stringer("db", table.DB.Name),
+				zap.Stringer("table", table.Info.Name),
+				zap.Uint64("expect tiflash replica", table.Info.TiFlashReplica.Count),
```
Comment (Leavrth) [inline br/pkg/restore/client.go:1979]: `table.Info.TiFlashReplica` may be nil, according to the above `if` compare.

**Review by Leavrth — APPROVED** [review]
rest LGTM

**Review by Leavrth — APPROVED** [review]

**Pull request merged**